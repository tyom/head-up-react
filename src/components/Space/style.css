.dashboard {
  height: 100vh;
  padding: 3px;
  display: flex;
  flex-flow: row wrap;
  overflow: auto;
  transition: 200ms;
}

/* States */

.is-active {}
