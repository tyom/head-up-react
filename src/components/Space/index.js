import React from 'react';
import PropTypes from 'prop-types';
import kebabCase from 'lodash/kebabCase';
import classNames from 'classnames';

import './style.css';

export default class Space extends React.Component {
  componentDidMount() {
    this.handleScrolling();
  }

  componentDidUpdate() {
    this.handleScrolling({ behavior: 'smooth' });
  }

  handleScrolling(behavior) {
    if (!this.props.isActive) {
      return;
    }
    this._dashboard.scrollIntoView && this._dashboard.scrollIntoView(behavior);
  }

  render() {
    const dashboardId = kebabCase(this.props.name);

    return (
      <div
        id={dashboardId}
        ref={c => (this._dashboard = c)}
        styleName={classNames('dashboard', {
          'is-active': this.props.isActive,
        })}
      >
        {this.props.children}
      </div>
    );
  }
}

Space.defaultProps = {
  isActive: false,
  cells: [],
};

Space.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.node,
  cells: PropTypes.array,
  isActive: PropTypes.bool,
};
