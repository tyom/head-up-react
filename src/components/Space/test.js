import React from 'react';
import Space from './';

const scrollIntoViewSpy = jest.fn();

function mountComponent(props) {
  const component = mount(<Space {...props}>content</Space>);
  const componentInstance = component.instance();

  componentInstance._dashboard = {
    scrollIntoView: scrollIntoViewSpy,
  };

  return componentInstance;
}

describe('Space: structure', () => {
  let rendered;

  afterEach(() => {
    expect(rendered).toMatchSnapshot();
  });

  test('default component', () => {
    rendered = render(<Space name="my dashboard">content</Space>);
  });

  test('active state', () => {
    rendered = render(
      <Space name="my dashboard" isActive={true}>
        Active dashboard
      </Space>
    );
  });
});

describe('Space: behaviour', () => {
  let componentInstance;

  describe('scrolling', () => {
    beforeEach(() => {
      componentInstance = mountComponent({
        name: 'my dashboard',
        isActive: true,
      });
    });

    afterEach(() => {
      jest.resetAllMocks();
    });

    test('scroll into view on mount', () => {
      componentInstance.componentDidMount();
      expect(scrollIntoViewSpy).toBeCalledWith(undefined);
    });

    test('scroll into view on update', () => {
      componentInstance.componentDidUpdate();
      expect(scrollIntoViewSpy).toBeCalledWith({ behavior: 'smooth' });
    });
  });

  test('scrolling only for active', () => {
    componentInstance = mountComponent({
      name: 'my dashboard',
      isActive: false,
    });

    componentInstance.componentDidUpdate();
    expect(scrollIntoViewSpy).not.toBeCalled();
  });
});
