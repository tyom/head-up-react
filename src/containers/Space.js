import { connect } from 'react-redux';

import { activateSpace } from '../actions';
import Space from '../components/Space';

const mapStateToProps = (state, ownProps) => ({
  children: state.spaces,
});

export const mapDispatchToProps = dispatch => ({
});

export default connect(mapStateToProps, mapDispatchToProps)(Space);
