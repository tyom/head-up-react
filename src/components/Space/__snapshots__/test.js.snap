// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space: structure active state 1`] = `
<div
  class="style_dashboard__1FOfF style_is-active__1dck2"
  id="my-dashboard"
>
  Active dashboard
</div>
`;

exports[`Space: structure default component 1`] = `
<div
  class="style_dashboard__1FOfF"
  id="my-dashboard"
>
  content
</div>
`;
